resource/origin.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resource
+
+import (
+	"github.com/juju/errors"
+)
+
+// These are the valid resource origins.
+const (
+	originUnknown Origin = iota
+	OriginUpload
+	OriginStore
+)
+
+var origins = map[Origin]string{
```

Comment:
Complete developer preference here, but I believe the stringer @natefinch demonstrated in the last sprint is *a* standard way to turn iotas into strings. Use this? If not, that's totally fine.